docs/contributing.rst

```diff
@@ -86,12 +86,42 @@ modules only, rather than individual classes or functions.
 Testing
 ^^^^^^^
 
-Twine is tested against Python versions 3.6, 3.7, and 3.8. To run these tests
-locally, you will need these versions of Python installed on your machine.
+We use `pytest`_ for writing and running tests.
```

Comment:
I feel like these docs, while they might help a new user, are largely general-purpose guidance. If it were me, to avoid having to keep these docs up-to-date, I'd direct users to a common place to learn about `tox` and `pytest` and maybe link to the places in the twine code where non-default behaviors are defined. More docs means more maintenance, so I aim to provide basic directions ("run tox") and leave it an an exercise for the user to follow the link to tox to learn how to install and use it. That said, there's nothing objectionable here.

Reply:
@jaraco I definitely didn't want to duplicate information from the official docs from the tools, but rather call out the places where twine has non-default behaviors that streamline testing workflow.

A middle-ground that might be more maintainable would be move to move some these examples to comments in `tox.ini`, but I feel like I don't see that done very often.